test/cbmc/proofs/IPUtils/usGenerateProtocolChecksum_IPv6/usGenerateProtocolChecksum_IPv6_harness.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * FreeRTOS memory safety proofs with CBMC.
+ * Copyright (C) 2022 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "queue.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_IP_Private.h"
+
+/* CBMC includes. */
+#include "cbmc.h"
+
+/* Function usGenerateChecksum is proven to be correct separately.
+ * Check if input buffer is readable. */
+uint16_t usGenerateChecksum( uint16_t usSum,
+                             const uint8_t * pucNextData,
+                             size_t uxByteCount )
+{
+    __CPROVER_assert( __CPROVER_r_ok( pucNextData, uxByteCount ), "pucNextData should be readable." );
+}
+
+void harness()
+{
+    size_t uxBufferLength;
+    uint8_t * pucEthernetBuffer;
+    BaseType_t xOutgoingPacket;
+    EthernetHeader_t * pxEthernetHeader;
+    IPPacket_IPv6_t * pxIPPacket;
+    uint16_t usHeaderLength;
+
+    /* The buffer must contain enough buffer size for ethernet header + IPv6 header and less than MTU size. */
+    __CPROVER_assume( uxBufferLength >= ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER + sizeof( ProtocolHeaders_t ) && uxBufferLength < ipconfigNETWORK_MTU );
+    pucEthernetBuffer = safeMalloc( uxBufferLength );
+    __CPROVER_assume( pucEthernetBuffer != NULL );
+
+    /* This test case verifies IPv6 only. */
+    pxIPPacket = ( IPPacket_IPv6_t * ) pucEthernetBuffer;
+    __CPROVER_assume( pxIPPacket->xEthernetHeader.usFrameType == ipIPv6_FRAME_TYPE );
+
+    /* Set to valid input. */
+    __CPROVER_assume( ( xOutgoingPacket == pdTRUE ) || ( xOutgoingPacket == pdFALSE ) );
```

Comment:
This can also be removed, assumption should be minimum i think

Reply:
Removed